spring-cloud-deployer-resource-support/src/main/java/org/springframework/cloud/deployer/resource/support/DelegatingResourceLoader.java

```diff
@@ -51,10 +87,25 @@ public Resource getResource(String location) {
 			Assert.notNull(scheme, "a scheme (prefix) is required");
 			ResourceLoader loader = this.loaders.get(scheme);
 			Assert.notNull(loader, String.format("no ResourceLoader available for scheme: %s", scheme));
-			return loader.getResource(location);
+
+			Resource resource = loader.getResource(location);
+			if (existsAsFile(resource)) {
+				return resource;
+			} else {
+				File cachedResource = new File(cacheDirectory, resource.getFilename());
```

Comment:
This has several problems IMHO:
* Same name files, but with different paths will collide (_e.g._ http://foo/bar/somefile.jar and http://woot/other/somefile.jar)
* What if the resource changes on the remote end?